jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -395,11 +400,30 @@ private static String startLocalAgent() throws IbmAttachOperationFailedException
 		} catch (Throwable e) {
 			throw new IbmAttachOperationFailedException("startLocalManagementAgent error starting agent:"+e.getClass()+" "+e.getMessage());		 //$NON-NLS-1$ //$NON-NLS-2$
 		}
-		String addr = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties().getProperty(IPC.LOCAL_CONNECTOR_ADDRESS);
-		if (null == addr) {
-			throw new IbmAttachOperationFailedException("startLocalManagementAgent: "+IPC.LOCAL_CONNECTOR_ADDRESS+" not defined");		 //$NON-NLS-1$ //$NON-NLS-2$
+		/*
+		 * sun.management.Agent.startLocalManagementAgent() in Java 8 sets 
+		 * c.s.m.j.localConnectorAddress in System.properties.
+		 * jdk.internal.agent.Agent.startLocalManagementAgent() in Java 9 sets 
+		 * c.s.m.j.localConnectorAddress in a different Properties object.
+		 */
+		Properties systemProperties = com.ibm.oti.vm.VM.getVMLangAccess().internalGetProperties();
+		String addr;
+		/*[IF Sidecar19-SE-B165]*/
+		synchronized (systemProperties) {
+			addr = systemProperties.getProperty(LOCAL_CONNECTOR_ADDRESS);
+			if (Objects.isNull(addr)) {
+				addr = VMSupport.getAgentProperties().getProperty(LOCAL_CONNECTOR_ADDRESS);
```

Comment:
Why does this probe both the old and new locations for the property?

Reply:
The intent of this is to set the local connector address in the system properties, per the reference implementation, and return the address.  

If the property is unset in the system properties, then the agent has not been started before.  This is not an error condition.  However, if the agent has been started  the property should have been set.